### ISSUE TITLE
While creating a query, a DOM node is immediately inserted

### ISSUE THREAD

**Issue by Janpot** [mui/mui-toolpad#1150]
### Duplicates

- [X] I have searched the existing issues

### Latest version

- [X] I have tested the latest version

### Current behavior 😯

While creating a new query, a default DOM node is immediately inserted in the application. This results in fetches to the backend before the user had the chance to edit the query. This is wasteful. Instead, on query creation the query should stay in draft, detached from the dom until the user presses "save" the first time.

Blocked on https://github.com/mui/mui-toolpad/issues/1126

### Context 🔦

Splitted off from https://github.com/mui/mui-toolpad/issues/1001 as being lower priority

**Comment by Ratikant-rout** [comment]
Can You Assign Me This Issue?

**Comment by prakhargupta1** [comment]
@Ratikant-rout While this issue is blocked on #1126 currently, feel free to pick any other issue.
And it is great to see your interest in the project.
